[yyanyee] iP

yyanyee opened the pull request:
### Clover

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Clover frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

#### Code Example:
```java
public class Clover {
    public static void main(String[] args) {
        System.out.println("Hello! I'm Clover");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yyanyee committed:
Level-1

yyanyee committed:
Level-1

yyanyee committed:
Level-2

yyanyee committed:
Level-3

yyanyee committed:
Level-3 (Updated)

yyanyee committed:
Level-4

yyanyee committed:
A-TextUiTesting

yyanyee committed:
Level-5

yyanyee committed:
Level-6

yyanyee committed:
Level-7

yyanyee committed:
Merge branch-Level-7 to master

yyanyee committed:
Level-8

yyanyee committed:
Merge branch-Level-8 to master

yyanyee committed:
A-MoreOOP

yyanyee committed:
A-Packages

yyanyee committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-Gradle

Gradle added to project

yyanyee committed:
A-Gradle

yyanyee committed:
Merge branch 'branch-A-Gradle'

Set up Gradle

yyanyee committed:
A-JUnit

yyanyee committed:
Merge branch 'branch-A-JUnit'

Added test files

yyanyee committed:
A-JavaDoc

yyanyee committed:
A-CodingStandard

yyanyee committed:
Level-9

LimErnest commented on src/main/java/clover/util/Parser.java:
```diff
@@ -0,0 +1,70 @@
+        String[] split = input.split(" ", 2);
+        String command = split[0].toLowerCase();
+
+        switch (command) {
```
Perhaps you can put the switch block into a try catch block and catch the respective exceptions so that u don't need to repeat the try catch block

LimErnest commented on src/main/java/clover/util/Storage.java:
```diff
@@ -0,0 +1,88 @@
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+
+            /*
```
Good use of comment to show what is the code in the file

LimErnest commented on src/main/java/clover/util/Storage.java:
```diff
@@ -0,0 +1,88 @@
+            String[] split = line.split(" \\| "); //split up line at even "|"
+            Task task;
+
+            switch (split[0]) {
```
Switch block has the wrong indenting, should be 

```
switch (split[0]) {
case
```

rithiklenin replied:
Good catch!

LimErnest commented on src/main/java/clover/util/Ui.java:
```diff
@@ -0,0 +1,75 @@
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + task);
+        System.out.println(" Now you have " + size + " tasks in the list.");
+        System.out.println("____________________________________________________________");
```
Maybe you can save a private final variable for the line and just call that variable instead of hardcoding the line

LimErnest commented on src/main/java/UnknownCommandException.java:
```diff
@@ -0,0 +1,5 @@
+public class UnknownCommandException extends Exception {
```
Perhaps add more Javadoc comments to explain more about the functionality of your code or the purpose of creating certain classes?

rithiklenin commented on src/main/java/clover/util/Parser.java:
```diff
@@ -0,0 +1,70 @@
+package clover.util;
+
+import clover.command.*;
```
You should list the imported classes explicitly

rithiklenin commented on src/main/test/java/seedu/clover/ParserTest.java:
```diff
@@ -0,0 +1,13 @@
+import clover.util.Parser;
+import clover.util.Parser.Commands;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
You should explicitly list it out as:

`import static org.junit.jupiter.api.Assertions.assertEquals;`

rithiklenin commented on src/main/test/java/seedu/clover/TaskTest.java:
```diff
@@ -0,0 +1,13 @@
+import clover.task.Task;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Same for this. You should explicitly list it out

rithiklenin commented on src/main/java/clover/util/Parser.java:
```diff
@@ -0,0 +1,70 @@
+        String command = split[0].toLowerCase();
+
+        switch (command) {
+            case "bye":
```
Switch block ("case") has the wrong indentation here as well.

rithiklenin reviewed (commented):
Overall well done. Just a few minor changes to do.

yyanyee committed:
Level-10

yyanyee committed:
Merge branch 'branch-Level-10'

Added GUI to Chat bot.

yyanyee committed:
A-Assertions

Add assertions to document assumptions in key methods

Task index must be valid before accessing tasks in the list.
Task descriptions should never be empty.

This change introduces assertions to document these assumptions and to
help catch potential bugs during runtime in debug mode. It improves
code safety and ensures edge cases are handled properly.

yyanyee committed:
A-Assertions

Add assertions to document assumptions in key methods

Task index must be valid before accessing tasks in the list.
Task descriptions should never be empty.

This change introduces assertions to document these assumptions and to
help catch potential bugs during runtime in debug mode. It improves
code safety and ensures edge cases are handled properly.

A-CodeQuality

Improve on code quality by refactoring methods to reduce duplication
and enhance readability.

Enumerations introduced in Parser class to reduce hardcoded values
and improve overall code quality.

yyanyee committed:
Merge pull request #1 from yyanyee/branch-A-Assertions

A-Assertions

yyanyee committed:
Merge branch 'master' into branch-A-CodeQuality

This merge integrates the latest changes from the master branch into
branch-A-Assertions. By keeping branch-A-Assertions up-to-date with
master, we ensure compatibility and prevent potential merge conflicts
in the future.

yyanyee committed:
A-CodeQuality-redo

yyanyee committed:
A-CodeQuality updated

yyanyee committed:
C-Help

yyanyee committed:
A-BetterGui

yyanyee committed:
Merge branch 'branch-A-BetterGui'
Improvement on Gui

Background image of chatbot added with improvements to the aesthetics
of the dialog box.

yyanyee committed:
A-Personality

yyanyee committed:
Merge branch 'branch-A-Personality'
Make chatbot more personalised.

Add welcome message to greet user when the application is opened.

yyanyee committed:
Merge branch 'branch-A-Personality'
Make chatbot more personalised.

Add welcome message to greet user when the application is opened.

yyanyee committed:
Add a representative screenshot of Barry

yyanyee committed:
Add README.md to document key features and usage

The project lacks clear documentation for users on how to interact with the chatbot,
making it harder to understand its features and capabilities.

Add a comprehensive README.md that includes a user guide, explanations for key
features, and examples of usage. This improves the user experience and provides
essential guidance on how to add tasks, deadlines, and use other commands.

The inclusion of examples and expected outcomes helps users get familiar
with the app's functionality quickly and accurately.

yyanyee committed:
Merge branch 'master' of https://github.com/yyanyee/ip into branch-A-UserGuide

yyanyee committed:
Merge branch 'master' of https://github.com/yyanyee/ip

Add README.md and user guide for Barry.

yyanyee committed:
Minor touch up on code and comments

yyanyee committed:
Clean up of code and comments

yyanyee committed:
Update README.md

yyanyee committed:
Update README.md

yyanyee committed:
Update README.md

yyanyee committed:
Edit README.md file

yyanyee committed:
Merge branch 'master' into branch-A-UserGuide

yyanyee committed:
Merge pull request #4 from yyanyee/branch-A-UserGuide

Updated README.md

yyanyee committed:
update tasks.txt

yyanyee committed:
Merge pull request #6 from yyanyee/branch-A-CodeQuality

Branch A code quality
